Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.h

```diff
@@ -52,70 +63,91 @@ class MediaPlayerPrivateWebM : public MediaPlayerPrivateInterface, public CanMak
     void load(const URL&, const ContentType&, MediaSourcePrivateClient&) final;
 #endif
 #if ENABLE(MEDIA_STREAM)
-    void load(MediaStreamPrivate&) final { };
+    void load(MediaStreamPrivate&) final;
 #endif
     
     void cancelLoad() final;
     
+    PlatformLayer* platformLayer() const final;
+    
     void play() final;
     void pause() final;
     
     FloatSize naturalSize() const final { return m_naturalSize; };
     
-    bool hasVideo() const final { return m_hasAudio; };
-    bool hasAudio() const final { return m_hasVideo; };
+    bool hasVideo() const final { return m_hasVideo; };
+    bool hasAudio() const final { return m_hasAudio; };
     
     void setPageIsVisible(bool) final;
     
+    MediaTime currentMediaTime() const final;
+    MediaTime durationMediaTime() const final { return m_duration; };
+    
     bool seeking() const final { return m_seeking; };
     
     bool paused() const final { return m_paused; };
     
     MediaPlayer::NetworkState networkState() const final { return m_networkState; };
     MediaPlayer::ReadyState readyState() const final { return m_readyState; };
     
-    std::unique_ptr<PlatformTimeRanges> buffered() const final { return makeUnique<PlatformTimeRanges>(); };
+    std::unique_ptr<PlatformTimeRanges> buffered() const final;
     
-    bool didLoadingProgress() const final { return false; };
+    bool didLoadingProgress() const final;
     
-    void paint(GraphicsContext&, const FloatRect&) final;
+    void paint(GraphicsContext&, const FloatRect&) final { };
     
     DestinationColorSpace colorSpace() final { return DestinationColorSpace::SRGB(); };
     
     void setNaturalSize(FloatSize);
     void setHasAudio(bool);
     void setHasVideo(bool);
+    void setDuration(MediaTime);
```

Comment:
`void setDuration(MediaTime&&)` or `void setDuration(const MediaTime&)` if it doesn't make sense to move the original.